scalameta/trees/shared/src/main/scala/scala/meta/internal/ast/InternalTrees.scala

```diff
@@ -44,6 +44,8 @@ trait InternalTree {
     if (privateParent != null) scala.Some(privateParent) else None
   }
 
+  def productFields: Seq[String]
```

Comment:
I'd like to highlight the fact that this method is accompanied by the more traditional `Product` methods, so that `InternalTrees.scala` looks like:

```scala
trait InternalTree extends Product {
  ...
  // NOTE: InternalTree inherits traditional productXXX methods from Product
  // and also adds a new method called productFields.
  // def productPrefix: String
  // def productArity: Int
  // def productElement(n: Int): Any
  // def productIterator: Iterator[Any]
  def productFields: Seq[String]
  ...
}
```